From f007a41cee4a722de89b0ec95de3e2fcb28960a2 Mon Sep 17 00:00:00 2001
From: Felix Fietkau <nbd@nbd.name>
Date: Thu, 16 Mar 2023 09:11:22 +0530
Subject: [PATCH 005/281] OpenWrt:
 100-compiler.h-only-include-asm-rwonce.h-for-kernel-code.patch

compiler.h: only include asm/rwonce.h for kernel code

This header file is not in uapi, which makes any user space code that includes
linux/compiler.h to fail with the error 'asm/rwonce.h: No such file or directory'

Fixes: e506ea451254 ("compiler.h: Split {READ,WRITE}_ONCE definitions out into rwonce.h")
Signed-off-by: Felix Fietkau <nbd@nbd.name>

Change-Id: I51a78faf776219a438aeca88e820934413b3b137
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
---
 include/linux/compiler.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/linux/compiler.h b/include/linux/compiler.h
index 973a1bfd7ef5..330784b5cf4f 100644
--- a/include/linux/compiler.h
+++ b/include/linux/compiler.h
@@ -203,6 +203,8 @@ void ftrace_likely_update(struct ftrace_likely_data *f, int val,
 	__v;								\
 })
 
+#include <asm/rwonce.h>
+
 #endif /* __KERNEL__ */
 
 /*
@@ -243,6 +245,4 @@ static inline void *offset_to_ptr(const int *off)
  */
 #define prevent_tail_call_optimization()	mb()
 
-#include <asm/rwonce.h>
-
 #endif /* __LINUX_COMPILER_H */
-- 
2.17.1

